src/hdmf/build/objectmapper.py

```diff
@@ -209,7 +209,12 @@ def convert_dtype(cls, spec, value):
             ret_dtype = tmp_dtype
         elif isinstance(value, AbstractDataChunkIterator):
             ret = value
-            ret_dtype, warning_msg = cls.__resolve_dtype(value.dtype, spec_dtype)
+            if spec_dtype is _unicode:
+                ret_dtype = "utf8"
+            elif spec_dtype is _ascii:
+                ret_dtype = "ascii"
+            else:
+                ret_dtype, warning_msg = cls.__resolve_dtype(value.dtype, spec_dtype)
         else:
```

Comment:
Why are _unicode and _ascii handled separately here rather than in ``__resolve_dtype``?

Reply:
__resolve_dtype is used for determining the precision of a _numerical dtype_, so it doesn't work for string types.